### ISSUE TITLE
Frameworks are duplicated when elasticsearch mesos scheduler restarted by marathon

### ISSUE THREAD

**Issue by garyelephant** [mesos/elasticsearch#530]
Frameworks are duplicated when elasticsearch mesos scheduler restarted by marathon.

I set --frameworkName elasticsearch_test1001 when starting elasticsearch. If I update some configuration of this application in marathon and redeploy(triggered automatically), another framework with same framework name, but with different framework id, is setup in mesos. You can check it in Mesos webUI like this:

![image](https://cloud.githubusercontent.com/assets/2291859/13841383/9c510aa8-ec63-11e5-8d46-9ab2c1789fad.png)

The release I used is : [elasticsearch-mesos-scheduler-1.0.0-withDependencies.jar](https://github.com/mesos/elasticsearch/releases/download/1.0.0/elasticsearch-mesos-scheduler-1.0.0-withDependencies.jar)

**Comment by philwinder** [comment]
Hi @garyelephant. This took me a while to figure out! So, when you do redeploy through marathon, it sends a SIGTERM to the framework. Which this framework considers a "normal" shutdown. I.e. it kills the instances and, crucially, deletes all zookeeper state. So when the framework creates a new instance, it sees that there is no zookeeper state and creates a new framework ID!

This was introduced because other people complained about left over zk state.

So, in short, this isn't an issue. If it is a problem for you, then kill the scheduler yourself with anything other than a sigterm and the zookeeper state will be preserve.

Thanks!

**Comment by garyelephant** [comment]
@philwinder So in my case, how to remove the other 2 framework, which is stale, from mesos?

**Comment by philwinder** [comment]
Hi Gary. I believe that Mesos also stores it's state in zookeeper. So either restart the master (which I think should flush out the state) or remove the offending state in zookeeper. You'll have to look this one up in the Mesos documentation.

**Comment by garyelephant** [comment]
@philwinder Thanks
